res/controllers/M-Audio-Xponent-Advanced-scripts.js

```diff
@@ -0,0 +1,842 @@
+
+    for (channel = 1; channel <= MaudioXponent.maxDecks; channel++) {
+        var group = "[Channel" + (channel) + "]";
+        var deck = MaudioXponent.decks[channel - 1];
+        deck.id = channel;
+        deck.beatState = false;
+        deck.group = group;
+        deck.progressMeterStatusByte = deck.isBankA ? 0xB3 : 0xB8;
+        deck.progressMeterSecondByte = deck.isLeft ? 0x14 : 0x15;
+        deck.vuMeterStatusByte = deck.isBankA ? 0xB3 : 0xB8;
+        deck.vuMeterSecondByte = deck.isLeft ? 0x12 : 0x13;
+        deck.scratchEnabled = false;
+        deck.scratching = false;
+        deck.shift = false;
+        deck.warnAt = 0;
+        deck.filterLow = 1;
+        deck.filterMid = 1;
+        deck.filterHigh = 1;
+        deck.pregain = 1;
+
+        engine.connectControl(group, "play_indicator", "MaudioXponent.onPlay");
+        engine.connectControl(group, "playposition", "MaudioXponent.onPlayPositionChange");
+        engine.connectControl(group, "duration", "MaudioXponent.onTrackLoaded");
+        engine.connectControl(group, "beat_active", "MaudioXponent.onBeatActive");
+        engine.connectControl(group, "eject", "MaudioXponent.onEject");
+
+        for (i = 1; i <= 5; i++) {
+            engine.connectControl(group, "hotcue_" + i + "_enabled", "MaudioXponent.onHotCue");
+        }
+
+        engine.connectControl(group, "loop_enabled", "MaudioXponent.onLoopExit");
+        engine.connectControl(group, "loop_in", "MaudioXponent.onLoopIn");
+        engine.connectControl(group, "loop_out", "MaudioXponent.onLoopOut");
+        for (i = 0.125; i < 16; i *= 2) {
+            engine.connectControl(group, "beatloop_" + i + "_enabled", "MaudioXponent.onBeatLoop");
+        }
+
+        engine.connectControl(group, "keylock", "MaudioXponent.onkeyLock");
+        engine.connectControl(group, "bpm", "MaudioXponent.onBpmChanged");
+        engine.connectControl(group, "pfl", "MaudioXponent.onPflChanged");
+
+        engine.connectControl(group, "reverse", "MaudioXponent.onReverse");
+        engine.connectControl(group, "reverseroll", "MaudioXponent.onReverse");
+        engine.connectControl(group, "rate_temp_down", "MaudioXponent.onNudge");
+        engine.connectControl(group, "rate_temp_up", "MaudioXponent.onNudge");
+
+        engine.connectControl("[EqualizerRack1_" + group + "_Effect1]", "button_parameter1", "MaudioXponent.onFilterKill");
+        engine.connectControl("[EqualizerRack1_" + group + "_Effect1]", "button_parameter2", "MaudioXponent.onFilterKill");
+        engine.connectControl("[EqualizerRack1_" + group + "_Effect1]", "button_parameter3", "MaudioXponent.onFilterKill");
+
+        engine.softTakeover(group, "rate", true);
+        engine.softTakeover(group, "volume", true);
+
+        // Soft-takeovers that aren't yet working correctly (Might need to save/restore on bank change)
+        engine.softTakeover(group, "filterLow", true);
```

Comment:
These probably aren't working because you're using the deprecated aliases. Since 2.0 the EQs are [special effects racks](http://mixxx.org/wiki/doku.php/mixxxcontrols#eqs_and_filters).